### PR TITLE
Fix overflow problem with failed resize on amp-list load-more

### PR DESCRIPTION
**Pull request by cathyxz**
Based on Ali's bug bash, contents reported at go/infinite-scroll-burndown: 
* **Fix**  load more does not show up when `amp-list` does not resize and therefore overflows bug. 
* **Fix** If initial "height" of amp-list is larger than the number of items shown, "Load more button" keeps showing up until height matches then infinite scroll kicks in bug.

This PR tries to solve the problem of cutting off the show-more button when a resize attempt is denied (i.e. when the bottom of the amp-list is in the top half of the viewport AND there is content below). The bug will put us in an awkward state where we can never load-more of anything. 

The UX in this proposed solution (below is non-ideal because it's overlaid on top of the amp-list contents. However, I'm not sure there's a way to force the resize to happen to make enough space for only the button, and I'm not 100% convinced that we should cause it to content jump. Basically this keeps parity with how we currently do overflow. 

screen shot 2019-01-04 at 2 44 38 pm

### CODE REVIEW

File: build-system/app.js [hunk 1]
```diff
@@ -1192,6 +1192,30 @@ const generateJson = numberOfItems => {
...
+};
+
+app.get('/infinite-scroll-random/:category', function(request, response) {
+  const {category} = request.params;
+  const result = generateResults(category);
+  result['load-more-src'] =
+      `/infinite-scroll-random/${category}?${randInt(10000)}`;
+
+  response.json(result);
+  response.end();
```
Comment (cvializ) [inline build-system/app.js]: nit: `response.json` calls `response.end` under the hood, so you can omit it.


> Remember, res.json() under the hood uses res.send() which in turn also uses res.end() to send a response and to end it. This is true for both Restify and Express.js as well.
> - https://fullstack-developer.academy/res-json-vs-res-send-vs-res-end-in-express/

File: build-system/app.js [hunk 2]
```diff
@@ -1192,6 +1192,30 @@ const generateJson = numberOfItems => {
...
+  }
+
+  r.items = items;
+  return r;
+};
+
+app.get('/infinite-scroll-random/:category', function(request, response) {
+  const {category} = request.params;
+  const result = generateResults(category);
+  result['load-more-src'] =
```
Comment (cvializ) [inline build-system/app.js]: Should this line be part of `generateResults`?

File: extensions/amp-list/0.1/amp-list.css [hunk 3]
```diff
@@ -46,6 +46,11 @@ amp-list[load-more] [load-more-button].i-amphtml-default-ui,
   text-align: center;
 }
 
+[load-more-button].i-amphtml-list-load-more-overflow.i-amphtml-default-ui {
```
Comment (aghassemi) [inline extensions/amp-list/0.1/amp-list.css]: this should not be specific to our `.i-amphtml-default-ui` class and should apply to all cases.

Comment (cathyxz) [reply extensions/amp-list/0.1/amp-list.css]: Done.

File: build-system/app.js (hunk 2 above)
Comment (cathyxz) [reply build-system/app.js]: Done

**Pull request merged**